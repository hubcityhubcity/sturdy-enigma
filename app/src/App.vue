<template>
  <div id="app" :class="themeClass">
    <div id="app-content">
      <div id="bar-container">
        <CookieBar v-if="!isMobileApp" />
      </div>
      <router-view />
    </div>
  </div>
</template>

<script>
import { mapState, mapGetters } from "vuex"
import AppHeader from "common/AppHeader"
import CookieBar from "common/CookieBar"
import DisconnectedBar from "common/DisconnectedBar"
import UserMenu from "account/UserMenu"
import MaintenanceBar from "common/MaintenanceBar"
import MobileMenu from "common/MobileMenu"
import NetworkSelector from "common/NetworkSelector"
import store from "./vuex/store"

export default {
  name: `app`,
  components: {
    CookieBar,
  },
  computed: {
    ...mapState([`session`]),
    isMobileApp() {
      return this.session.mobile
    },
    themeClass() {
      return `lunie-light`
    },
  },
  store,
}
</script>

<style>
@import "./styles/app.css";
</style>
