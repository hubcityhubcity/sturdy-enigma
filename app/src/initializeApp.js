/* istanbul ignore file: really just integrations */

import { checkForNewLunieVersions } from "scripts/check-for-new-lunie-versions"
import {
  enableGoogleAnalytics,
  setGoogleAnalyticsPage,
} from "scripts/google-analytics"
import config from "src/../config"
import Router, { routeGuard } from "./router"
import Store from "./vuex/store"

if (navigator && navigator.serviceWorker) {
  // remove any existing service worker
  navigator.serviceWorker.getRegistrations().then(function (registrations) {
    for (let registration of registrations) {
      registration.unregister()
    }
  })
}

export const bootError = (error) => {
  document.querySelector("#app").innerHTML = `
  <div class="boot-error">
    <h2>Uh oh. Lunie failed to start. Our team has been notified of the issue. Please try refreshing this page in a few minutes. If the error persists please contact us at http://help.lunie.io/</h2>

    <p class="error">${error}</p>
  </div>
    `
}

export default async function init(urlParams, env = process.env) {
  // add error handlers in production
  if (env.NODE_ENV === `production`) {
    // check every minute if new Lunie versions have been deployed
    checkForNewLunieVersions()
    enableGoogleAnalytics(config.google_analytics_uid)
  }

  const store = Store()

  const router = Router(store)
  setGoogleAnalyticsPage(router.currentRoute.path)

  store.dispatch(`loadLocalPreferences`)
  // store.dispatch(`checkForPersistedAddresses`)

  return { store, router }
}
