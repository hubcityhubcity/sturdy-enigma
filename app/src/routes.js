/**
 * In this module we took care of the definition of our routes, with parameters, children and component related to them
 * @module routes
 */

/**
 * Routes are all defined here
 */
/* istanbul ignore file */
export default (store) => {
  return [
    {
      path: `/reveal/:address`,
      name: `reveal`,
      component: require("account/RevealSeedModal").default,
    },
    {
      path: `/delete/:address/:addressNetworkId`,
      name: `delete`,
      component: require("account/ForgetAccountModal").default,
    },
    {
      path: `/`,
      name: `manage-accounts-modal`,
      component: () => import(`./components/account/ManageAccountsModal`),
      meta: {
        networkSpecificRoute: true,
      },
    },
  ]
}

// handle direct sign in from the extension via deeplink
export async function extensionSignIn({ to, next }, store) {
  let network = store.getters.networks.find(
    ({ id }) => id === to.params.network
  )

  await store.dispatch(`signIn`, {
    sessionType: `extension`,
    address: to.params.address,
    networkId: to.params.network,
  })
  next(`/${network.slug}/portfolio`)
}
