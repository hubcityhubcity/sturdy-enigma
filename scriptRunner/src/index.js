var express = require('express')
var bodyParser = require('body-parser')
var timeout = require('connect-timeout');
var morgan = require('morgan')
var app = express()
require('dotenv').config()
const config = require('../config')

// Constants
const PORT = process.env.PORT || 9000;
const HOST = process.env.NODE_ENV = "docker" ? '0.0.0.0' : 'localhost';

const polkadotrewards = require("./polkadotrewards")

app.use(bodyParser.json())
app.use(timeout(120000))
app.use(morgan('combined'))
app.use(function (req, res, next) {
    const authenticationToken = req.header("Authorization")
    if (authenticationToken !== config.authenticationToken) {
        res
            .status(403)
            .send()
        return
    }
    next()
})


app.post('/polkadotrewards', polkadotrewards)

app.listen(PORT, HOST, () => console.log(`Script server running on http://localhost:${PORT}`))