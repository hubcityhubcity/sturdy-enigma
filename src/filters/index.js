import { percent } from "../scripts/num"
import moment from "moment"

export const date = date => moment(date).format("MMMM Do YYYY, HH:mm")

export const fromNow = date => moment(date).fromNow()

export const noBlanks = function(value) {
  return value === undefined ||
    value === null ||
    value === `` ||
    value === `[do-not-modify]`
    ? `--`
    : value
}

export const percentOrPending = function(value, totalValue, pending) {
  return pending ? `--` : percent(totalValue === 0 ? 0 : value / totalValue)
}

export const formatBech32 = (address, longForm = false, length = 4) => {
  if (!address) {
    return `Address Not Found`
  } else if (address.indexOf(`1`) === -1) {
    return `Not A Valid Bech32 Address`
  } else if (longForm) {
    return address
  } else {
    return address.split(`1`)[0] + `â€¦` + address.slice(-1 * length)
  }
}

export const resolveValidatorName = (address, validators) => {
  if (validators[address]) {
    return validators[address].name
  }
  return formatBech32(address)
}
