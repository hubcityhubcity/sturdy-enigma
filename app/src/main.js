/* istanbul ignore file */

import Vue from "vue"
import VTooltip from "v-tooltip"
import Vuelidate from "vuelidate"
import VueClipboard from "vue-clipboard2"
import { focusParentLast } from "src/directives"
import App from "./App.vue"
import init, { bootError } from "./initializeApp"
import { getURLParams } from "scripts/url"
import { Plugins } from "@capacitor/core"
import config from "src/../config"
import * as Sentry from "@sentry/browser"
import * as Integrations from "@sentry/integrations"
import "material-design-icons-iconfont/dist/material-design-icons.css"
import AsyncComputed from "vue-async-computed"

if (config.sentryDSN) {
  Sentry.init({
    dsn: config.sentryDSN,
    integrations: [new Integrations.Vue({ Vue, attachProps: true })],
    release: process.env.LUNIE_VERSION,
  })
}

Vue.config.productionTip = false

Vue.use(VTooltip)

Vue.use(Vuelidate)
Vue.use(VueClipboard)
Vue.use(AsyncComputed)

Vue.directive(`focus-last`, focusParentLast)

const urlParams = getURLParams(window)
init(urlParams)
  .then(({ store, router }) => {
    const { SplashScreen, StatusBar } = Plugins

    new Vue({
      router,
      ...App,
      store,
      mounted() {
        if (config.mobileApp) {
          SplashScreen.hide()
          StatusBar.show()
        }
      },
    }).$mount("#app")
  })
  .catch((error) => {
    Sentry.captureException(error)
    bootError(error)
  })
