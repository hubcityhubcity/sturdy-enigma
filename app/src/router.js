import router from "vue-router"
import routes from "./routes"
import Vue from "vue"

/* istanbul ignore next */
Vue.use(router)

/* istanbul ignore next */
const Router = (store) =>
  new router({
    mode: process.env.VUE_APP_E2E ? undefined : "history",
    routes: routes(store),
  })

export default Router
