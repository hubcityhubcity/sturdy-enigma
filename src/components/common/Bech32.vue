<template>
  <div class="bech32-address">
    <div
      v-tooltip.top="`Click to copy`"
      v-clipboard:copy="address"
      v-clipboard:success="() => onCopy()"
      class="address"
    >
      {{ address | formatBech32(longForm) }}
    </div>
    <div :class="{ active: copySuccess }" class="copied">
      <i class="material-icons notranslate">check</i>
    </div>
  </div>
</template>

<script>
import { formatBech32 } from "src/filters"

export default {
  name: `bech32-address`,
  filters: {
    formatBech32
  },
  props: {
    address: {
      type: String,
      required: true
    },
    longForm: {
      type: Boolean,
      required: false,
      default: false
    }
  },
  data: () => ({
    copySuccess: false
  }),
  methods: {
    onCopy() {
      this.copySuccess = true
      setTimeout(() => {
        this.copySuccess = false
      }, 2500)
    }
  }
}
</script>
<style>
.bech32-address {
  align-items: flex-start;
  display: inline-flex;
  padding: 0;
  margin: 0;
  font-size: inherit;
}

.bech32-address .address {
  color: var(--link);
  cursor: pointer;
  white-space: nowrap;
}

.bech32-address .address:hover {
  color: var(--link-hover);
}

.bech32-address .copied {
  align-items: flex-end;
  display: flex;
  font-size: 10px;
  opacity: 0;
  padding-left: 0.25rem;
  padding-top: 2px;
  transition: opacity 500ms ease;
}

.bech32-address .copied.active {
  opacity: 1;
}

.bech32-address .copied i {
  color: var(--success);
  font-size: 12px;
  padding-bottom: 2px;
  padding-right: 0;
}
</style>
