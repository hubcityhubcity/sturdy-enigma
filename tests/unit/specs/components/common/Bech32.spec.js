import { shallowMount, createLocalVue } from "@vue/test-utils"
import VueClipboard from "vue-clipboard2"
import Bech32 from "src/components/common/Bech32"
import { formatBech32 } from "src/filters"

const localVue = createLocalVue()
localVue.directive(`clipboard`, VueClipboard)
localVue.directive(`tooltip`, () => {})
localVue.filter(`formatBech32`, formatBech32)

describe(`Bech32`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Bech32, {
      localVue,
      propsData: { address: `cosmosftw123456789` },
      data: () => ({
        copySuccess: false
      })
    })
  })

  it(`should show a short address`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should show a long address`, () => {
    wrapper.setProps({ longForm: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should return 'address not found'`, () => {
    wrapper.setProps({ address: null })
    expect(wrapper.find(".address").text()).toBe(`Address Not Found`)
  })

  it(`should return a validation error message`, () => {
    wrapper.setProps({ address: `cosmosaddress2asdfasdfasdf` })
    expect(wrapper.find(".address").text()).toBe(`Not A Valid Bech32 Address`)
  })

  it(`should return a short address with everything before the 1`, () => {
    wrapper.setProps({ address: `cosmosaddress1asdfasdfasdf` })
    expect(wrapper.find(".address").text()).toBe(`cosmosaddressâ€¦asdf`)
  })

  it(`should return a long address when long-form set`, () => {
    wrapper.setProps({ address: `cosmosaddress1asdfasdfasdf`, longForm: true })
    expect(wrapper.find(".address").text()).toBe(`cosmosaddress1asdfasdfasdf`)
  })

  describe(`onCopy`, () => {
    it(`should set and reset copySuccess`, () => {
      jest.useFakeTimers()
      wrapper.vm.onCopy() // old test style to make timer work
      expect(wrapper.vm.copySuccess).toBe(true)

      jest.runAllTimers()
      expect(wrapper.vm.copySuccess).toBe(false)
    })
  })

  // TODO: not sure how to test the v-clipboard directive events
  xit(`clicking copy copies the address`, () => {
    jest.useFakeTimers()
    expect(
      wrapper
        .find(`.copied`)
        .classes()
        .includes(`active`)
    ).toBe(false)

    wrapper.find(`.address`).trigger(`click`)
    expect(
      wrapper
        .find(`.copied`)
        .classes()
        .includes(`active`)
    ).toBe(true)

    jest.runAllTimers()
    expect(
      wrapper
        .find(`.copied`)
        .classes()
        .includes(`active`)
    ).toBe(false)
  })
})
