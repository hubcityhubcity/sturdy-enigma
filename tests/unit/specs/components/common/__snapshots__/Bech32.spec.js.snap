// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bech32 should show a long address 1`] = `
<div
  class="bech32-address"
>
  <div
    class="address"
  >
    
    cosmosftw123456789
  
  </div>
   
  <div
    class="copied"
  >
    <i
      class="material-icons notranslate"
    >
      check
    </i>
  </div>
</div>
`;

exports[`Bech32 should show a short address 1`] = `
<div
  class="bech32-address"
>
  <div
    class="address"
  >
    
    cosmosftwâ€¦6789
  
  </div>
   
  <div
    class="copied"
  >
    <i
      class="material-icons notranslate"
    >
      check
    </i>
  </div>
</div>
`;
